### PR TITLE
MBS-10556 add missing retries to ReportApi and FileStorage

### PR DESCRIPTION
**Pull request by dsvoronin**
All methods are POST, was retrying only GET's
discovered while browsing kibana logs

### CODE REVIEW

File: subprojects/common/report-viewer/src/main/kotlin/com/avito/report/ReportsApi.kt [hunk 1]
```diff
@@ -66,15 +67,21 @@ interface ReportsApi : ReportsAddApi, ReportsFetchApi {
...
                         verbose = verboseHttp,
-                        logger = loggerFactory.create<ReportsApi>(),
+                        logger = logger,
                         readTimeoutSec = readTimeout,
-                        writeTimeoutSec = writeTimeout
+                        writeTimeoutSec = writeTimeout,
+                        retryInterceptor = RetryInterceptor(
+                            logger = logger,
+                            allowedMethods = listOf("POST")
+                        )
```
Comment (eugene-krivobokov) [inline subprojects/common/report-viewer/src/main/kotlin/com/avito/report/ReportsApi.kt:84]: What statuses did we receive from RV?
Do we need to add them to the interceptor?

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/common/report-viewer/src/main/kotlin/com/avito/report/ReportsApi.kt (hunk 1 above)
Comment (dsvoronin) [reply subprojects/common/report-viewer/src/main/kotlin/com/avito/report/ReportsApi.kt:84]: https://github.com/avito-tech/avito-android/blob/0bada247948b9db5ff93a89b33251b730ab65bcb/subprojects/common/okhttp/src/main/kotlin/com/avito/http/RetryInterceptor.kt#L18-L22

These defaults are ok, it's 502 mainly

File: build-logic/testing-convention/src/main/kotlin/convention.unit-testing.gradle.kts [hunk 2]
```diff
@@ -19,6 +19,15 @@ tasks.withType<Test>().configureEach {
...
     jvmArgs = listOf("--add-opens", "java.base/java.lang=ALL-UNNAMED")
+
+    /**
+     * IDEA adds an init script, using it to define if it is an IDE run
+     * used in `:test-project`
+     */
+    systemProperty(
+        "isInvokedFromIde",
+        gradle.startParameter.allInitScripts.find { it.name.contains("ijtestinit") } != null
+    )
```
Comment (eugene-krivobokov) [inline build-logic/testing-convention/src/main/kotlin/convention.unit-testing.gradle.kts:30]: What about Gradle tests?

File: subprojects/common/report-viewer/src/test/kotlin/com/avito/report/ReportsApiTest.kt [hunk 3]
```diff
@@ -55,7 +62,9 @@ internal class ReportsApiTest {
 
     @Test
     fun `getReport - returns Error - when throws exception with no data`() {
-        mockWebServer.enqueue(MockResponse().setResponseCode(500))
+        repeat(retryCount) {
```
Comment (eugene-krivobokov) [inline subprojects/common/report-viewer/src/test/kotlin/com/avito/report/ReportsApiTest.kt:65]: Is it an imitation of "unrecoverable" case?
In this case we duplicate implementation here. We can show the intention by Dispatcher and don't rely on count.

**Pull request merged**